### PR TITLE
Run BK integration tests on PRs and release branches

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
Runs the new integration testing infrastructure provisioned by buildkite and terraform (ESS)

Integration tests take 1 h 10 minutes instead of ~1:30. 

Independent test groups execution allows to retry tests independently with a dedicated ESS stack:
Screenshot 2024-11-27 at 16 43 16

[Build example 1](https://buildkite.com/elastic/elastic-agent-extended-testing/builds/4907#01936da4-9be4-446b-a01c-d1721b3c21c6)
[Build example 2](https://buildkite.com/elastic/elastic-agent-extended-testing/builds/4908)
[Build example 3](https://buildkite.com/elastic/elastic-agent-extended-testing/builds/4920)
[Build example 4](https://buildkite.com/elastic/elastic-agent-extended-testing/builds/4939)

#### What groups do we run
- [x] default
- [x] container
- [x] fleet-upgrade-to-pr-build
- [x] upgrade
- [x] fleet
- [x] fqdn
- [x] deb
- [x] fleet-airgapped
- [x] fleet-privileged
- [x] fleet-airgapped-privileged
- [x] rpm

#### What OS do we run
- [x] Windows 2022
- [x] Ubuntu 24.04 x86_64
- [x] Ubuntu 24.04 ARM (temporarily disabled due to a [known issue](https://github.com/elastic/ingest-dev/issues/4610))
- [x] RHEL 8 (rpm)

By comparing test reports, I verified that we run the same test sets for groups. Placed the tests in groups in this [spreadsheet](https://docs.google.com/spreadsheets/d/1kuExKnVyjiwpGSl_YUn_enVVwg-XuMnu2wWMMud-sn4/edit?gid=0#gid=0)

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by blakerouse — CHANGES_REQUESTED** [review]
This cut over cannot occur until this pipeline is generated.

File: docs/test-framework-dev-guide.md
Comment (rdner) [inline docs/test-framework-dev-guide.md:324]: Not sure I understood the sentence though.

```suggestion
  Add the new group to the `matrix` in the corresponding steps. The matrix is a list of all test groups that are executed in the step. 
```

**Review by ycombinator — APPROVED** [review]
LGTM once @rdner's typo fix [suggestion](https://github.com/elastic/elastic-agent/pull/6148#discussion_r1862430076) is accepted.  I'll merge the PR then.

**Comment by ycombinator** [comment]
Chatted with @rowlandgeoff about merging this PR.  He mentioned that there might be a couple of other fixes in other PRs that need to go in first before this PR can be merged.  So @pazone, just ping me on this PR whenever this is ready for re-review / merge. Thanks!

**Comment by rowlandgeoff** [comment]
Pavel and I have had overlapping PTO, but from what I can tell these issues need resolution before this cutover PR is merged:
- https://github.com/elastic/ingest-dev/issues/4603 (appears fixed already)
- https://github.com/elastic/ingest-dev/issues/4610 (fix ready to merge)
- https://github.com/elastic/ingest-dev/issues/4671 (fix required to ensure parity of test debugging experience)

**Review by rdner — APPROVED** [review]

**Comment by rowlandgeoff** [comment]
> Pavel and I have had overlapping PTO, but from what I can tell these issues need resolution before this cutover PR is merged:
> 
> * [[CI][Elastic-agent] BK integration tests: fix sceduled test run failures ingest-dev#4603](https://github.com/elastic/ingest-dev/issues/4603) (appears fixed already)
> * [[Elastic-agent] ARM integration tests execution causes occasional test failures ingest-dev#4610](https://github.com/elastic/ingest-dev/issues/4610) (fix ready to merge)
> * [[CI] [Elastic-agent] Add diagnostics artifacts to buildkite integration tests ingest-dev#4671](https://github.com/elastic/ingest-dev/issues/4671) (fix required to ensure parity of test debugging experience)

We're good on these prerequisite issues. The ARM integration tests are not a blocker for cutover since they have been disabled for now. We'll get those working after the cutover.

**Review by rowlandgeoff — APPROVED** [review]
I reviewed the PR and the example pipeline runs with Pavel. Looks good! Once merged, We will kick off several pipeline runs to test it out and watch all runs very closely.

**Pull request merged**
